pkg/controller/installation/core_controller.go

```diff
@@ -790,6 +765,29 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 		}
 	}
 
+	freshInstall := reflect.DeepEqual(status, operator.InstallationStatus{})
+	if freshInstall {
+		// This is a fresh deployment. Check if an API server has been requested. If it has, we should
+		// wait for it to roll out successfully before proceeding, allowing for configuration to be
+		// specified prior to calico/node launching.
+		apiserver, msg, err := utils.GetAPIServer(ctx, r.client)
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				reqLogger.V(5).Info("failed to get APIServer CR", "err", err)
+				r.status.SetDegraded(msg, err.Error())
+				return reconcile.Result{}, err
+			}
+		}
+		if apiserver.Status.State != operator.TigeraStatusReady {
+			// API server is not yet ready. Wait for it.
+			r.status.SetDegraded("Waiting for API server to be ready", "")
+			return reconcile.Result{RequeueAfter: 30 * time.Second}, nil
```

Comment:
I was uncomfortable watching the APIServer because it would be a new dependency for this controller - i.e., a new version of the operator would fail unless the APIServer CRD was also present.

I suppose it's not the end of the world, but thought this would be more friendly / backwards compatible. I don't think we're saying the APIServer is required for OSS just yet?